### ISSUE TITLE
Support additional arguments in nn.Identity.forward

### ISSUE THREAD

**Issue by PaulLerner** [pytorch/pytorch#42015]
*Apologies if this was already discussed, I was not able to find any similar issue*

## 🚀 Feature
Support additional arguments in `nn.Identity.forward`

i.e.

```py
def forward(self, input: Tensor, *args, **kwargs) -> Tensor:
    return input
```

instead of

```py
def forward(self, input: Tensor) -> Tensor:
    return input
```
## Motivation

Correct me if I'm wrong but I guess the point of the `Identity` module is to be able to make highly modular models by "replacing" some given module by the Identity module.

Or, since some modules take more arguments than just `input` (e.g. `TransformerEncoder` takes `mask` and `src_key_padding_mask`), they cannot be "replaced" by Identity.

cc @albanD @mruberry

**Comment by mruberry** [comment]
Thanks for the suggestion, @PaulLerner! This sounds good to me. User's can work around this issue today by replacing the module's forward call, but that, as @PaulLerner says, kinda defeats the point.

@albanD would there be an issues with forward taking (*args, **kwargs) that you see? Wasn't there some issue with hooks and kwargs?

**Comment by albanD** [comment]
Hi,

The two questions I would have are:
- What is the use case for you? In particular, what is before/after this module? The Sequential module does not support such things (only one Tensor as input).
- How do you return the `kwargs` in a good way? If you eat up the args/kwargs like in your example, then it's not really an identity anymore.

**Comment by PaulLerner** [comment]
Hello,

My use case is fine-tuning BERT, either by:
- keeping BERT frozen and adding a `TransformerEncoder` after it (`self.encoder=TransformerEncoder(...)`)
- keeping BERT trainable but add nothing after it (`self.encoder=Identity(...)`)

You have a good point, in my use case/example I only return the input but then it's not really an identity anymore.

**Comment by blakedewey** [comment]
+1 for this issue. I have the use case where the Module I am looking to replace takes an additional argument as a condition, e.g.

```
class AdaptiveInstanceNorm2d(nn.Module):
    def __init__(self, num_ch, cond_ch):
        super().__init__()
        self.norm = nn.InstanceNorm2d(num_ch, affine=False)
        self.gamma_fc = nn.Linear(cond_ch, num_ch)
        self.beta_fc = nn.Linear(cond_ch, num_ch)

    def forward(self, in_tensor: torch.Tensor, cond_tensor: torch.Tensor):
        x = self.norm(in_tensor)
        gamma = self.gamma_fc(cond_tensor).unsqueeze(-1).unsqueeze(-1).expand_as(x)
        beta = self.beta_fc(cond_tensor).unsqueeze(-1).unsqueeze(-1).expand_as(x)
        return (x * gamma) + beta
```
In this case, I cannot replace with `nn.Identity` because the forward call has two inputs. Using `*args` would solve this issue.

**Comment by Laubeee** [comment]
@PaulLerner how did you solve this to mark it as completed? did you implement your own dummy-Module?

**Comment by PaulLerner** [comment]
Hi, I marked solved because the issue is old.

At the time I did as in my above comment:

```py
def forward(self, input: Tensor, *args, **kwargs) -> Tensor:
    return input
```
